Server/APIs/AuthenticationAPI.py

```diff
@@ -0,0 +1,64 @@
+from flask import request, jsonify, Blueprint, current_app
+from werkzeug.security import generate_password_hash, check_password_hash
+import uuid
+import jwt
+from datetime import datetime, timedelta
+from functools import wraps
+from Model.Player import Player
+from Server.Database.DBMethods import DBMethods
+from Common.AuthenticationErrors import AuthenticationErrors
+
+
+authentication_api = Blueprint('authentication_api', __name__)
+
+
+def token_required(func):
+    @wraps(func)
+    def decorated(*args, **kwargs):
+        token = None
+        if 'x-access-token' in request.headers:
+            token = request.headers['x-access-token']
+        if not token:
+            return jsonify({'message': 'Token is missing !!'}), 401
+        try:
+            data = jwt.decode(token, current_app.config['SECRET_KEY'])
```

Comment:
Amazing usage of a decorator. 
if you dont know what a is decorator read about that, ]
really an advanced python subject!